lib/matplotlib/patches.py

```diff
@@ -3204,23 +3204,116 @@ class _Curve(_Base):
+            angleA : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            angleB : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            scaleA : float, default *mutation_size*
+                The mutation_size for the beginning bracket
+            scaleB : float, default *mutation_size*
+                The mutation_size for the end bracket
             """
+
             self.head_length, self.head_width = head_length, head_width
+            self.widthA, self.widthB = widthA, widthB
+            self.lengthA, self.lengthB = lengthA, lengthB
+            self.angleA, self.angleB = angleA, angleB
+            self.scaleA, self.scaleB = scaleA, scaleB
+
+            self._beginarrow_head = False
+            self._beginarrow_bracket = False
+            self._endarrow_head = False
+            self._endarrow_bracket = False
+
+            _split_arrow = self.arrow.split("-", 1)
+
+            if _split_arrow[0] == "<":
+                self._beginarrow_head = True
+                self._beginarrow_bracket = False
+            elif _split_arrow[0] == "<|":
+                self._beginarrow_head = True
+                self._beginarrow_bracket = False
+                self.fillbegin = True
+            elif (_split_arrow[0] == "]" or
+                  _split_arrow[0] == "|"):
+                self._beginarrow_head = False
+                self._beginarrow_bracket = True
+            elif self.beginarrow is True:
+                self._beginarrow_head = True
+                self._beginarrow_bracket = False
+
+                _api.warn_deprecated('3.5', message="The use of"
+                                     "beginarrow is now decapitated, prefer"
+                                     "using 'arrow' for the same parameter"
+                                     "result")
+            elif self.beginarrow is False:
+                self._beginarrow_head = False
+                self._beginarrow_bracket = False
+
+                _api.warn_deprecated('3.5', message="The use of"
+                                     "beginarrow is now decapitated, prefer"
+                                     "using 'arrow' for the same parameter"
+                                     "result")
+
+            if _split_arrow[len(_split_arrow) - 1] == ">":
+                self._endarrow_head = True
+                self._endarrow_bracket = False
+            elif _split_arrow[len(_split_arrow) - 1] == "|>":
+                self._endarrow_head = True
+                self._endarrow_bracket = False
+                self.fillend = True
+            elif (_split_arrow[len(_split_arrow) - 1] == "[" or
+                  _split_arrow[len(_split_arrow) - 1] == "|"):
```

Comment:
You passed `maxsplit`; there are only two things in `_split_arrow`:
```suggestion
            if _split_arrow[1] == ">":
                self._endarrow_head = True
                self._endarrow_bracket = False
            elif _split_arrow[1] == "|>":
                self._endarrow_head = True
                self._endarrow_bracket = False
                self.fillend = True
            elif _split_arrow[1] in ("[", "|"):
```